tests/stepDefinitions/loginContext.js

```diff
@@ -1,35 +1,39 @@
 const { Given, When, Then } = require('@cucumber/cucumber')
 const { expect } = require("@playwright/test");
-
-
-// for valid scenario
-Given('user {string} has navigated to the admin login page', function (string) {
-  console.log(" Given ");
+const loginBtnLocator=".btn.btn-block.btn-primary"
+const emailFieldLocator="#spree_user_email"
+const passwordFieldLocator="#spree_user_password"
+const dashboardSelector="//*[@id='missingHomepage']/div/div[2]/h4[1]"
+const invalidErrorSelector=".alert-danger"
+
+Given('user {string} has navigated to the admin login page', async function (string) {
+  await page.goto("http://127.0.0.1:3000/admin/login")
+  await expect(page.url()).toBe("http://127.0.0.1:3000/admin/login");
 });
 
-When('user {string} logs in with following credentials', function (string, dataTable) {
-  console.log(" When ");
+When('user {string} logs in with following credentials',async function (string, dataTable) {
   for (const { email, password } of dataTable.hashes()) {
-    console.log(email, password)
+    await page.locator(emailFieldLocator).fill(email);
```

Comment:
Just a suggestion!!!
You can use this alternative as well
```suggestion
    await page.fill(emailFieldLocator, email);
```